### PR TITLE
Add caching for CF plans

### PR DESCRIPTION
**Pull request by dotchev**
Load all service plans from CF between brokers and visibility resync
Use this data to look up in memory:
* All the plans from SM-managed brokers during visibility resync
* The plan guid from broker name and catalog id during visibility update (save 3 calls to CF per visibility)

Update in-memory data on each broker change

Depends on https://github.com/Peripli/service-broker-proxy/pull/122

### CODE REVIEW

**Comment by dotchev** [comment]
Executing E2E tests...

File: cf/cfmodel/plan_resolver.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+			BrokerName:    broker.Name,
+			CatalogPlanID: plan.UniqueId,
+			Public:        plan.Public,
+		})
+	}
+}
+
+// ResetBroker replaces the data for a particular broker
+func (r *PlanResolver) ResetBroker(brokerName string, plans []cfclient.ServicePlan) {
+	r.mutex.Lock()
```
Comment (NickyMateev) [inline cf/cfmodel/plan_resolver.go]: You can just call `r.DeleteBroker(brokerName)` here?

File: cf/service_access.go [hunk 2]
```diff
@@ -101,19 +104,25 @@ func (pc *PlatformClient) updateOrgVisibilityForPlan(ctx context.Context, plan c
...
+
 	if plan.Public == isPublic {
 		return nil
 	}
-	_, err := pc.UpdateServicePlan(plan.Guid, ServicePlanRequest{
+
+	_, err := pc.UpdateServicePlan(plan.GUID, ServicePlanRequest{
 		Public: isPublic,
 	})
+	if err != nil {
```
Comment (NickyMateev) [inline cf/service_access.go]:
```go
if _, err := pc.UpdateServicePlan(plan.GUID, ServicePlanRequest{ Public: isPublic }); err != nil {
   return err
}
```

File: cf/cfmodel/plan_resolver.go (hunk 1 above)
Comment (NickyMateev) [reply cf/cfmodel/plan_resolver.go]: Nevermind, you need the lock

**Comment by dotchev** [comment]
E2E tests passed

**Pull request merged**